extensions/vertx-web/deployment/src/main/java/io/quarkus/vertx/web/deployment/VertxWebProcessor.java

```diff
@@ -468,8 +489,16 @@ private String generateHandler(HandlerDescriptor desc, BeanInfo bean, MethodInfo
                     .setModifiers(ACC_PRIVATE | ACC_FINAL);
         }
 
-        implementConstructor(bean, invokerCreator, beanField, contextField, containerField);
-        implementInvoke(desc, bean, method, invokerCreator, beanField, contextField, containerField, transformedAnnotations,
+        FieldCreator validatorField = null;
+        if (desc.isProducedResponseValidated()) {
+            // If the produced item needs to be validated, we inject the Validator
+            validatorField = invokerCreator.getFieldCreator("validator", Methods.VALIDATION_VALIDATOR)
+                    .setModifiers(ACC_PUBLIC | ACC_FINAL);
```

Comment:
The field is accessed from another class (a callback). Made it public final after discussion with @mkouba